src/grpc/grpc_server.cc

```diff
@@ -1186,6 +1186,20 @@ CommonHandler::RegisterModelStatistics()
               res.mutable_empty_response()->set_ns(val);
             }
 
+            {
+              triton::common::TritonJson::Value stat_json;
+              err = res_json.MemberAsObject("cancel", &stat_json);
+              GOTO_IF_ERR(err, earlyexit);
+
+              uint64_t val;
+              err = stat_json.MemberAsUInt("count", &val);
+              GOTO_IF_ERR(err, earlyexit);
+              res.mutable_cancel()->set_count(val);
+              err = stat_json.MemberAsUInt("ns", &val);
+              GOTO_IF_ERR(err, earlyexit);
+              res.mutable_cancel()->set_ns(val);
+            }
```

Comment:
optional opportunity for cleanup while you're working on this part of the code, can be a separate PR if it interests you

---

Do you think all of these verbose blocks could be reduced to something like:
```cpp
void SetLatencyCountNs(std::string stat_name, <some_type> mutable_stat) {
              triton::common::TritonJson::Value stat_json;
              err = res_json.MemberAsObject(stat_name, &stat_json);
              GOTO_IF_ERR(err, earlyexit);

              uint64_t val;
              err = stat_json.MemberAsUInt("count", &val);
              GOTO_IF_ERR(err, earlyexit);
              mutable_stat->set_count(val);
              err = stat_json.MemberAsUInt("ns", &val);
              GOTO_IF_ERR(err, earlyexit);
              mutable_stat->set_ns(val);
}

void RegisterModelStatistics() {
...
  // Re-use helper instead of all these copy/pasted blocks
  SetLatencyCountNs("compute_infer", res.mutable_compute_infer());
  SetLatencyCountNs("compute_output", res.mutable_compute_output());
  ...
  SetLatencyCountNs("cancel", res.mutable_cancel());
}

```

Reply:
[Refactor RegisterModelStatistics()](https://github.com/triton-inference-server/server/pull/6904/commits/55ffd8eb11c2e305100eb47daa7bc4e1e73102b5)